tests/runner/test_runner.py

```diff
@@ -752,6 +757,87 @@ async def test_runner_executes_flow_runs(
             },
         ]
 
+    async def test_runner_does_not_duplicate_heartbeats(
+        self,
+        prefect_client: PrefectClient,
+        asserting_events_worker: EventsWorker,
+    ):
+        """
+        Regressiong test for issue where multiple invocations of `execute_flow_run`
```

Comment:
i almost wanna leave it in
```suggestion
        Regression test for issue where multiple invocations of `execute_flow_run`
```